cli/resources.rs

```diff
@@ -236,22 +235,28 @@ impl Read for Resource {
   }
 }
 
-impl AsyncRead for Resource {
-  fn poll_read(&mut self, buf: &mut [u8]) -> Poll<usize, Error> {
+pub trait DenoAsyncRead {
```

Comment:
Could you add a triple slash comment here describing the purpose of DenoAsyncRead (same for DenoAsyncWrite)

Something like this: /// DenoAsyncRead is the same as the AsyncRead trait but uses an ErrBox error instead of std::io:Error